src/Adapter/Product/ProductDataProvider.php

```diff
@@ -162,4 +163,32 @@ public function getImage($id_image)
             'base_image_url' => _THEME_PROD_DIR_ . $imageData->getImgPath(),
         ];
     }
+
+    /**
+     * Gets price with tax.
+     *
+     * @param int $productId
+     *
+     * @return float
+     */
+    public function getPriceWithTax(int $productId): float
+    {
+        return Product::getPriceStatic(
+            $productId,
+            true,
+            null,
+            (int) Configuration::get('PS_PRICE_DISPLAY_PRECISION'),
```

Comment:
Please use ConfigurationAdapter

Reply:
I guess this could be done in separated PR to move everything to DI?

Reply:
Not agree, adding this method should be done in correct way.

Reply:
but this also is related to `Context` class used inside this class, right? :) we should use `LegacyContext`?

Reply:
:+1:

Reply:
Changed to use ConfigurationAdapter, but I am not sure if changing whole class to use LegacyContext is a good idea for this PR as it would increase testing scope and potentially affect many other places.

Reply:
As we can see changing Configuration affects other places too (PHPStan is not happy about it). This is why I suggested to do it in separated PR, `ProductDataProvider` now has breaking changes

Reply:
Turns out it affects a lot of places. ProductDataProvider is used like this "(new ProductDataProvider())" so changing __construct breaks a lot of things. 

I could still use Adapter\Configuration without construct, but it introduces the same issue: if somebody updates construct for Adapter/Configuration it would break ProductDataProvider. 

So IMO it's best to use legacy classes here and as @kpodemski refactor the whole ProductDataProvider in separate PR.

Reply:
I agree with @JevgenijVisockij : we are dealing with complex logic here 😅  let's do it step by step, little by little, and with scoped PRs. @Progi1984 we aim to deliver the new Product Page Lists with multiple PRs 😉  so there will be iterations. But we want to avoid massive PRs as we know this makes review and QA hard, especially since we deal with a sensible part of the BO 😅

Reply:
cc @Progi1984 about `ConfigurationAdapter`